warp10/src/main/java/io/warp10/script/MemoryWarpScriptStack.java

```diff
@@ -564,165 +563,200 @@ public void exec(String line, long lineNumber) throws WarpScriptException {
-        statements = UnsafeString.split(line, ' ');
-      } else {
-        // We're either in multiline mode or the line had no whitespace inside
-        statements = new String[1];
-        if (inMultiline.get()) {
-          // If the line only contained the end of multiline indicator with possible wsp on both sides
-          // then set the statement to that, otherwise set it to the raw line
-          if(WarpScriptStack.MULTILINE_END.equals(line)) {
-            statements[0] = line;
+      if (inMultiline.get()) {
+        line = line.trim();
+        // End of multiline        
+        if (WarpScriptStack.MULTILINE_END.equals(line)) {
+          inMultiline.set(false);
+          String mlcontent = multiline.toString();
+          if (null != secureScript) {
+            secureScript.append(" ");
+            secureScript.append("'");
+            try {
+              secureScript.append(WarpURLEncoder.encode(mlcontent, StandardCharsets.UTF_8));
+            } catch (UnsupportedEncodingException uee) {
+            }
+            secureScript.append("'");
           } else {
-            statements[0] = rawline;
+            if (macros.isEmpty()) {
+              this.push(mlcontent);
+            } else {
+              macros.get(0).add(mlcontent);
+            }
           }
+          multiline.setLength(0);
         } else {
-          statements[0] = line;
+          // Append current line to existing multiline 
+          if (multiline.length() > 0) {
+            multiline.append("\n");
+          }
+          multiline.append(rawline);
+        }
+        return;
+      } else if (inComment.get()) {
+        int end = line.indexOf(COMMENT_END);
+        if (-1 == end) {
+          return; // No end of comment in this line, skip it
+        } else {
+          start = end; // Skip the beginning of the line, before */
         }
       }
-
-      //
-      // Report progress
-      //
-
-      progress();
-
+      
       //
-      // Loop over the statements
+      // Process line character by character, looking at block comments, comments, strings, then process statements.
       //
-
-      for (int st = 0; st < statements.length; st++) {
-        handleSignal();
-
-        String stmt = statements[st];
-
-        try {
-          //
-          // Skip empty statements if we are not currently building a multiline
-          //
-
-          if (0 == stmt.length() && !inMultiline.get()) {
-            continue;
+      String stmt;
+      int end = 0;
+      int pos = 0;
+      try {
+        for (pos = start; pos < line.length(); pos++) {
+
+          if (line.charAt(pos) <= ' ') {
+            continue; // Ignore spaces (or other control char)
           }
 
           //
-          // Trim statement
+          // Start of comment block /*
           //
-
-          if (!inMultiline.get()) {
-            stmt = stmt.trim();
+          if (line.length() - pos > 1 && line.charAt(pos) == '/' && line.charAt(pos + 1) == '*') {
```

Comment:
Comment start MUST either be at the beginning of the line or after a whitespace as the following is not valid WarpScript:

42/* .... */

Reply:
The way it is done, no problem. in your example, at pos = 0, we see a statement `42/*`, so it will fail as the previous parser will.